### ISSUE TITLE
[ possible ] Bug: Failing catching rapid clicking events on iOS 15?

### ISSUE THREAD

**Issue by mark-night** [facebook/react#22521]
React version:
v17.0.2

## Steps To Reproduce

1. Simple clicking event handling.
2. Rapid click on iOS 15 or later.

Link to code example:
[A simplified example](https://codesandbox.io/s/intelligent-cache-js9gz) to play with.
Just open the link on an iPhone upgraded to iOS 15 or later, and try rapid clicking letters to see the issue, some clicks are not handled without any feedback. (Rapid clicking on the same letter doesn't seem to have the issue.)

In comparison, [this is the same app with vanilla javascript](https://codesandbox.io/s/awesome-elbakyan-7tmn0), this works just fine without any issue. (So I'm guessing this might be ReactJS related.)

## The current behavior
Rapid clicking on DOM elements, not all clicks are caught and handled. ONLY ON IOS 15 OR LATER.

I've posted this on [stackoverflow](https://stackoverflow.com/questions/69473540/react-js-fails-catching-rapid-click-events-in-ios-15) too. Not sure if this is a ReactJS bug, or just Apple messed things again. 

Any suggestions where to look to find a fix? Thanks.

**Comment by mark-night** [comment]
@SubhayanDas08 Hi, thanks for checking into this. What I should do to "assign" the issue to you?

I also posted this question on stackoverflow, there is no any response/confirm so far. I even doubted could it be my miss-understanding on how to use React event listener setup. I reviewed my example codes, really couldn't discover anything suspicious. 

What's sad is I don't even have a clue where/how to start investigating this issue, as it seems only happen on iPhone browsers, (possible iPad as well but I don't have an iPad to test with), while on desktop, it seems to work just fine. (*Well, I'm not very sure if it's fine on desktop, because it's possible that by clicking with a mouse, it's hard to do a "rapid click", so it's hard to be revealed if it has issue too...*) 

I do need to mention though, per my observation, this only happens on iPhone upgraded to iOS 15 or later. I'm guessing there are some changes in iOS 15 that don't work well with React JS...
